### ISSUE TITLE
MAX - Guns no longer cause "puff" graphics when hitting materials?

### ISSUE THREAD

**Issue by fearlesswee** [TheGameCreators/GameGuruRepo#4729]
Like last report, ran through canyon offensive level to test another issue, and ran into this one. It appears that the player's weaponry does appropriately leave bullet holes, but the actual "puff" graphic (a puff of smoke on stone, blood on enemies, sparks on metal, etc.) no longer appear. Strangely, it would appear that the water splash DOES appear, and is the only material-based "puff" effect that still works.

(Also, this is more of a feature request, but these puff effects should seriously render much further out. They're such cheap little sprite effects, I can't imagine why they're culled so harshly as to only render within like, 10 feet of the player. Enemies should also make these puffs as well to add more feedback to a gun fight imo.)

Running the latest experimental build as of 6/23/2023

**Comment by davetgc** [comment]
I think the overall plan is to replace these with much improved particle effects.

**Comment by fearlesswee** [comment]
> I think the overall plan is to replace these with much improved particle effects.

YES. PLEASE. THANK YOU.

The rest of the industry moved on to particle effects over 20 years ago, I've been hollering for this change since day one! (#1692 )

But since this is no doubt some ways away, for now I think the puffs should at least be fixed, and made to render at much further distances than they currently do.

**Comment by LeeBamberTGC** [comment]
@fearlesswee I dropped in a boulder and the magnum, then shot it, and it shows the material puff of stone. Can you tell me which object you where shooting that did not produce a reaction as that would indicate a material issue:
![image](https://github.com/TheGameCreators/GameGuruRepo/assets/35840583/80473b3d-1155-4206-96d3-04a394113b3c)

**Comment by GatoBaguette** [comment]
Some objects (especially third party) might have "materialindex" set to "-1" which makes them not use any effect when hit. The only way to fix it is to go to each FPE of the object and set it to anywhere from 0 to 18 for instance (each number represents type of material)

**Comment by fearlesswee** [comment]
I wanted to clarify that when I did have this issue, it was on EVERY material and object, including ones that used to work, and the terrain itself. However, it seems to have somehow fixed itself with the latest experimental updates, so maybe the puff graphic files somehow were lost/deleted(?) and they were reacquired when MAX updated. Closing this since the issue mysteriously resolved itself lol

**Comment by fearlesswee** [comment]
As mentioned in issue #4728 , this issue has resurfaced with enough playtime. Notes on this issue reposted:

> Played through Canyon Offensive again, the issue didn't happen at first, but as I progressed through the level the enemies eventually lost their muzzle flash again, and, interestingly enough, at the same time bullet "puff" effects stopped working, meaning the two are likely connected.

> Can confirm this has not been fixed in latest exp build.

**Comment by fearlesswee** [comment]
@LeeBamberTGC Of course, I've been rallying for that change for about 2 years now! #1692 (The gargantuan effort was made to implement the AGKS particle system, but the last little 5% to actually hook it up to bullet hits, blood splatters, explosions, etc. wasn't finished!)

I do sincerely hope this can come sooner rather than later, it's definitely one of the most antiquated and (no offense) frankly ugly parts of MAX's presentation/visuals currently. The explosions in particular, since they don't properly "glow" (they get shadows casted onto them; in dark rooms they're hardly visible), and they can noticeably clip with nearby objects/entities due to their size making the "cardboard cutout" look way more apparent. 

I don't have a VR headset (yet), but I can only imagine how much more apparent their flat-ness is in 3D!
